client/src/components/feedback/feedback.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
+import { Container } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import FeedbackSend from './send';
+import FeedbackComment from './comment';
+import '../../styles/feedback.css';
+import Api from '../../services/api';
+
+class Feedback extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      allFeedback: [],
+      pieceFeedback: [],
+      maxCommentShow: 5,
+      start: 0,
+      last: 0,
+      firstLoad: false,
+    };
+    this.refresh = this.refresh.bind(this);
+  }
+
+  async componentDidMount() {
+    const { firstLoad, maxCommentShow, last } = this.state;
+    const { id } = this.props;
+    const data = await Api.getAllRestaurantFeedback('feedback?restaurantId=', id);
+
+    this.setState(() => {
+      if (data == null) {
+        return null;
+      }
+      data.reverse();
+      if (!firstLoad) {
+        return {
+          allFeedback: data,
+          pieceFeedback: data.slice(0, maxCommentShow),
+        };
+      }
+      return {
+        allFeedback: data,
+        pieceFeedback: data.slice(0, last + 1),
+      };
+    });
+  }
+
+  onClickShowMoreComment() {
+    const {
+      firstLoad, allFeedback, last, maxCommentShow,
+    } = this.state;
+    if (!firstLoad) {
+      this.setState({
+        last: last + maxCommentShow,
+        firstLoad: true,
+      });
+    }
+    if (allFeedback.length >= last + maxCommentShow) {
+      this.setState((prevState) => {
+        return {
+          pieceFeedback: prevState.allFeedback.slice(prevState.start,
+            prevState.last + prevState.maxCommentShow),
+          last: prevState.last + prevState.maxCommentShow,
+        };
+      });
+    } else if (allFeedback.length - last > 0) {
+      this.setState((prevState) => {
+        return {
+          pieceFeedback: prevState.allFeedback.slice(prevState.start,
+            prevState.last + prevState.maxCommentShow),
+          last: allFeedback.length,
+        };
+      });
+    }
+  }
+
+  onClickShowLessComment() {
+    const { maxCommentShow } = this.state;
+    this.setState((prevState) => {
+      if (prevState.last - prevState.maxCommentShow > maxCommentShow) {
+        return {
+          pieceFeedback: prevState.allFeedback.slice(prevState.start,
+            prevState.last - prevState.maxCommentShow),
+          last: prevState.last - prevState.maxCommentShow,
+        };
+      }
+      return {
+        last: 5,
+        pieceFeedback: prevState.allFeedback.slice(prevState.start, 5),
+      };
+    });
+  }
+
+  refresh() {
+    this.componentDidMount();
+  }
+
+  render() {
+    const stateFeed = this.state;
+    return (
+      <Container className="feedback">
+        <FeedbackSend id={this.props.id} refreshed={this.refresh} />
```

Comment:
Is id here a restaurant id? if so, you would be enough id={this.props} like in the line 26 (const { id } = this.props;)

Reply:
Unfortunately, it doesn't work with only this.props. I tried twice. Could you try on your device?

At all it should work if only we catch id in FeedbackSend later. But it is the same code just in another component.